repos/system_upgrade/common/actors/localrepos/libraries/localrepos.py

```diff
@@ -0,0 +1,92 @@
+import os
+
+from leapp.libraries.stdlib import api
+
+HOST_ROOT_MOUNT_BIND_PATH = '/installroot'
+LOCAL_FILE_URL_PREFIX = 'file://'
+
+
+def _adjust_local_file_url(repo_file_line):
+    """
+    Adjusts a local file url to the target user-space container in a provided
+    repo file line by prefixing host root mount bind '/installroot' to it
+    when needed.
+
+    :param str repo_file_line: a line from a repo file
+    :returns str: adjusted line or the provided line if no changes are needed
+    """
+    adjust_fields = ['baseurl', 'mirrorlist']
+
+    if LOCAL_FILE_URL_PREFIX in repo_file_line:
+        entry_field, entry_value = repo_file_line.strip().split('=', 1)
+        if not any(entry_field.startswith(field) for field in adjust_fields):
+            return repo_file_line
+
+        entry_value = entry_value.strip('\'\"')
+        path = entry_value[len(LOCAL_FILE_URL_PREFIX):]
+        new_entry_value = LOCAL_FILE_URL_PREFIX + os.path.join(HOST_ROOT_MOUNT_BIND_PATH, path.lstrip('/'))
+        new_repo_file_line = entry_field + '=' + new_entry_value
+        return new_repo_file_line
+    return repo_file_line
+
+
+def _extract_repos_from_repofile(context, repo_file):
+    """
+    Generator function that extracts repositories from a repo file in the given context
+    and yields them as list of lines that belong to the repository.
+
+    :param context: target user-space context
+    :param str repo_file: path to repository file (inside the provided context)
+    """
+    with context.open(repo_file, 'r') as rf:
+        repo_file_lines = rf.read().split('\n')
+
+    if repo_file_lines == ['']:
```

Comment:
Thinking about this condition again, I changed it entirely because if there were only multiple blank lines in a repo file, these lines would be treated as a repository. I changed it so that the lines before the first repoid are skipped and added a test for that. And I also incorporated the ``readlines()`` which made the unit tests bit more simple, contrary to what I thought when I decided to not use it before.